087-adopt-connect-health-endpoint.md

```diff
@@ -0,0 +1,86 @@
+
+This is the default HTTP probe configuration shared by both components:
+
+| Property name       | Default value | Description                                                                                  |
+|---------------------|---------------|----------------------------------------------------------------------------------------------|
+| initialDelaySeconds | 60            | The initial delay before first the health is first checked.                                  |
+| timeoutSeconds      | 5             | The timeout for each attempted health check.                                                 |
+| periodSeconds       | 10            | How often to perform the probe.                                                              |
+| successThreshold    | 1             | Minimum consecutive successes for the probe to be considered successful after having failed. |
+| failureThreshold    | 3             | Minimum consecutive failures for the probe to be considered failed after having succeeded.   |
+
+Strimzi does not support the HTTPs protocol for Kafka Connect REST API ([KIP-208](https://cwiki.apache.org/confluence/display/KAFKA/KIP-208%3A+Add+SSL+support+to+Kafka+Connect+REST+interface)).
+
+Example output with `/` endpoint (probes only care about the HTTP response status code):
+
+```sh
+$ kubectl exec my-cluster-kafka-0 -- curl -s http://my-connect-cluster-connect-api:8083 | jq
+{
+  "version": "3.9.0",
+  "commit": "a60e31147e6b01ee",
+  "kafka_cluster_id": "_xIIeVIOQMKXimv_l96WtQ"
+}
+```
+
+## Motivation
+
+Using the `/` endpoint for Kafka Connect and MirrorMaker 2 health checks is a common approach, but it does not actually test for readiness, because these requests can be completed before the worker is started.
+Instead, the `/health` endpoint waits for the worker startup phase to complete, which is made of internal topics creation if they do not exists, internal topics full read, and cluster join.
+
+If the worker has not yet completed the startup phase, or it is unable to respond in time, the response will have a 5xx status code.
+All Kafka Connect endpoints have a timeout of 90 seconds, with the exception of the `/health` endpoint where it is hardcoded to 10 seconds for faster unhealty worker detection.
+Unlike the `/` endpoint, the `/health` endpoint response message includes error details that can help with troubleshooting.
+
+Example output with `/health` endpoint (probes only care about the HTTP response status code):
+
+```sh
+$ kubectl exec my-cluster-kafka-0 -- curl -s http://my-connect-cluster-connect-api:8083/health | jq
+{
+  "status": "healthy",
+  "message": "Worker has completed startup and is ready to handle requests."
+}
+```
+
+## Proposal
+
+Strimzi currently supports both Kafka 3.8 and 3.9 releases.
+
+As the `/health` endpoint was introduced in Kafka 3.9, we will wait for Kafka 3.8 to go out of support in Strimzi before doing the switch.
```

Comment:
Maybe you can specifically mention Strimzi 0.46 as that is where the bulk of the Kafka 4.0 and ZooKeeper removal changes will land.